### ISSUE TITLE
Remember login amongst refresh

### ISSUE THREAD

**Issue by techjason** [tylerjrichards/st-paywall#23]
Would it be possible to implement remember login amongst refreshing? I have been experimenting with streamlit cookies packages but no luck with any of them. Great work btw!

**Comment by tylerjrichards** [comment]
this is a great point, I was chatting with @blackary about some ideas for this! it is trickier than we expected but hopefully we can figure out some solution for this

**Comment by pcejrowski** [comment]
I have gotten this to work with a cookie manager instead of the session state.
Example managers:
* https://pypi.org/project/streamlit-cookies-manager/
* https://github.com/Mohamed-512/Extra-Streamlit-Components#cookie-manager

**Comment by tylerjrichards** [comment]
sorry for the late reply here, but I think there are some big problems with this approach. Mainly, if you stick the email in the cookie, then anyone can edit the cookie on their end to an email that they know is subscribed, and get access to your app. So what you actually have to do is create some sort of a hash, and store that has in a database in the app, which is much trickier than we/I had originally thought

**Comment by tylerjrichards** [comment]
Nope, no movement on this @jasonmcbride. PRs are welcome though if you'd like to take a hack at it!

**Comment by jozkotrapko** [comment]
> I have gotten this to work with a cookie manager instead of the session state. Example managers:
> 
> * https://pypi.org/project/streamlit-cookies-manager/
> * https://github.com/Mohamed-512/Extra-Streamlit-Components#cookie-manager
> 
> I used the first one within this PR: #32

Would you be able to share a code, how did you manage to make it work? It would be very helpful! Thank you in advance.

**Comment by baogorek** [comment]
Hello, I just discovered this issue. Happy to work on it though I'm not an authentication expert.

Is there anything we can learn quickly from [Streamlit-Authenticator](https://github.com/mkhorasani/Streamlit-Authenticator}?
